[WIP] 4934 – Analytics Widget – Number

vasconsaurus opened the pull request:
## Description

This implements the number widget component. The number widget allows us to show a single important data point.

#### Highlights
- Title is customizable
- Main color is customizable via CSS var
- The widget will always contain a number
- The unit that goes with the main number is optional
- There is an optional “context” text that appears below the main number in case we want to add clarifying details
- The “empty” state is to replace the number with a hyphen -

#### Images
<img width="1392" alt="Screenshot 2024-09-16 at 11 21 08" src="https://github.com/user-attachments/assets/f8b91816-e605-4668-bf51-2ce90bff12cf">
<img width="965" alt="Screenshot 2024-09-16 at 11 21 20" src="https://github.com/user-attachments/assets/b9bf27f5-ef38-4da9-a84c-ba7ab1f618d8">

References: 4934

## Type of change

- [ ] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Security mitigation or enhancement
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

## How has this been tested?

I tested by adding it to the Sandbox and manually testing it there. And by adding a unit test:

```
npm run test:unit NumberWidget.test.js
```

## Things to pay attention to during code review

Please describe parts of the change that require extra attention during code review, for example:

- File FFFF, line LL: This refactoring does this and this. Is it consistent with how it’s implemented elsewhere?
- Etc.

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I've made sure my branch is runnable and given good testing steps in the PR description
- [ ] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [ ] If my components involve user interaction - specifically button, text fields, or other inputs - I have added a [BEM-like class name](https://meedan.atlassian.net/wiki/spaces/ENG/pages/1327628289/Naming+conventions+for+interactive+elements) to the element that is interacted with
- [ ] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)
- [ ] If I touched a file that included an eslint-disable-file header, I updated the code such that the disabler can be removed

vasconsaurus commented:
@danielevalverde I started to add the styles, it is still not finished, BUT could you take a look at the code just to see if it is going in the correct direction?

amoedoamorim commented on src/app/components/cds/charts/NumberWidget.test.js:
```diff
@@ -0,0 +1,29 @@
+    unit: null,
+  };
+
+  it('renders without crashing if languages is null', () => {
```
what languages? :sweat_smile:

amoedoamorim replied:
only nit I found to pick, otherwise this is looking great, Manu! :tada:

vasconsaurus replied:
> what languages? 😅

whooops 😅 copy and paste failed me!

vasconsaurus commented:
- I updated the propTypes to be `node` instead `string`.
- Added `<FormattedMessage />` to the Sandbox example
- Updated `Title` to be required, and updated test accordingly

amoedoamorim reviewed (approved):
Wooo! Awesome work, Manu! :tada:
